pkg/utils/controller/placement_resolver.go

```diff
@@ -0,0 +1,89 @@
+/*
+Copyright 2025 The KubeFleet Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controller
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	fleetv1beta1 "github.com/kubefleet-dev/kubefleet/apis/placement/v1beta1"
+	"github.com/kubefleet-dev/kubefleet/pkg/scheduler/queue"
+)
+
+const (
+	// namespaceSeparator is the separator used between namespace and name in placement keys.
+	namespaceSeparator = "/"
+)
+
+// FetchPlacementFromKey resolves a PlacementKey to a concrete placement object that implements PlacementObj.
+func FetchPlacementFromKey(ctx context.Context, c client.Client, placementKey queue.PlacementKey) (fleetv1beta1.PlacementObj, error) {
+	keyStr := string(placementKey)
+
+	// Check if the key contains a namespace separator
+	if strings.Contains(keyStr, namespaceSeparator) {
+		// This is a namespaced ResourcePlacement
+		parts := strings.Split(keyStr, namespaceSeparator)
+		if len(parts) != 2 {
+			return nil, NewUnexpectedBehaviorError(fmt.Errorf("invalid placement key format: %s", keyStr))
+		}
+
+		namespace := parts[0]
+		name := parts[1]
+
+		rp := &fleetv1beta1.ResourcePlacement{}
+		key := types.NamespacedName{
+			Namespace: namespace,
+			Name:      name,
+		}
+
+		if err := c.Get(ctx, key, rp); err != nil {
```

Comment:
need to wrap the error as the api server error.

so that In the caller, we don't need to wrap the error in scheduler.go line 139

placement, err := controller.FetchPlacementFromKey(ctx, s.client, placementName)
if err != nil {
		// Wrap the error for metrics; this method does not return an error.
		klog.ErrorS(controller.NewAPIServerError(true, err), "Failed to get placement", "placement", placementRef)

Reply:
this will break the following scheduler code 

if errors.IsNotFound(err) {